src/Datadog.Trace/Agent/Jaeger/JaegerExporter.cs

```diff
@@ -0,0 +1,158 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Runtime.CompilerServices;
+using System.Threading.Tasks;
+using Datadog.Trace.Logging;
+using Thrift.Protocol;
+using Thrift.Transport;
+
+namespace Datadog.Trace.Agent.Jaeger
+{
+    internal class JaegerExporter : IExporter
```

Comment:
Questions about actual and expected behaviour.

What happens if the trace endpoint becomes unavailable? For example what happens when:
- send a few traces [1]
- start the the OTel collector (or Jager if you send traces directly to Jaeger)
- send a few traces [2]
- stop the OTel collector
- send a few traces [3]
- start the the OTel collector
- send a few traces [4]

What traces are being transmitted after such test case? Only [2] and [4]? All of them?

1. Are the traces lost that were sent when the endpoint was down?
2. Are the new traces send to the endpoint after the endpoint is up again?

What is the expected behaviour in this scenario? Do the OTel specs say something about it?

I suggest testing it on Linux/MacOS and Windows the Sockets work differently on those OSes.

Reply:
Looking at the OTel .NET SDK it seems like the traces that failed to send return `ExportResult.Failure`, but both the SimpleProcessor and BatchProcessor (that call exporter.Export) do not look at the ExportResult that's returned, so success and failure is treated the same, which results in "dropping" the traces.

Reply:
If I understand it correctly this means that traces [2] and [4] should be sent. The rest should be "dropped".

Reply:
Yes, that's what I think should happen, and that would mean that our behavior would be consistent for now at least.

Reply:
So far I can confirm under Windows it works.

Reply:
Approving as "working on Windows" is already something worth merging in my opinion. Still, it should be tested on Linux or MacOS and another PR if there is any issue. I just remember that you got some problems with tests under Linux/MacOS and I think something bad may happen on these OSes.

Reply:
Tried with my ASP.NET Core sample app under Ubuntu LTS 20.04 and works as well.